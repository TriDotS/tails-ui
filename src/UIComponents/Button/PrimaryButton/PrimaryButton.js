import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  &:focus {
    outline: none;
  }
`

const PrimaryButton = ({ children, outline, fullWidth }) => (
  <Button
    className={
      outline
        ? `${fullWidth &&
            'w-full'} bg-transparent hover:bg-teal text-teal-dark font-semibold hover:text-white py-2 px-4 border-2 border-teal hover:border-transparent rounded inline-flex items-center`
        : `${fullWidth &&
            'w-full'} bg-teal hover:bg-teal-dark text-white font-sans font-bold py-2 px-4 shadow rounded inline-flex items-center`
    }
  >
    {children}
  </Button>
)

PrimaryButton.propTypes = {
  children: PropTypes.node,
  outline: PropTypes.bool,
  fullWidth: PropTypes.bool
}

export default PrimaryButton
