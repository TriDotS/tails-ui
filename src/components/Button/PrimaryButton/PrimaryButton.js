import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  &:focus {
    outline: none;
  }
`

const PrimaryButton = ({ children, outline }) => (
  <Button
    className={
      outline
        ? 'bg-transparent hover:bg-blue text-blue-dark font-semibold hover:text-white py-2 px-4 border-2 border-blue hover:border-transparent rounded'
        : 'bg-blue hover:bg-blue-dark text-white font-sans font-bold py-2 px-4 shadow rounded'
    }
  >
    {children}
  </Button>
)

PrimaryButton.propTypes = {
  children: PropTypes.node,
  outline: PropTypes.bool
}

export default PrimaryButton
