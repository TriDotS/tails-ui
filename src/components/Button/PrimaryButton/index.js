import PropTypes from 'prop-types'
import React from 'react'

const PrimaryButton = ({ children }) => (
  <button className="bg-blue hover:bg-blue-dark text-white font-bold py-2 px-4 rounded">
    {children}
  </button>
)

PrimaryButton.propTypes = {
  children: PropTypes.node
}

export default PrimaryButton
