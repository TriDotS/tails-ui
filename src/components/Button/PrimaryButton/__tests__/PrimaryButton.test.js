import PrimaryButton from '../PrimaryButton'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<PrimaryButton />, div)
})

test('primary button snapshot', () => {
  const tree = renderer.create(<PrimaryButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
