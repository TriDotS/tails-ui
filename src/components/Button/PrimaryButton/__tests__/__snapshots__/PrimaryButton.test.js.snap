// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primary button snapshot 1`] = `
<button
  className="bg-blue hover:bg-blue-dark text-white font-sans font-bold py-2 px-4 shadow rounded sc-bdVaJa igkfrU"
/>
`;
